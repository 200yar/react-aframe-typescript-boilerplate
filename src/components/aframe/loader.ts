export const loaderComponentConfig =  {

  schema: {'default': ''},

  /**
   * Do not use arrow function here
   * "this: any" is needed for the typescript compiler
   * https://github.com/Microsoft/TypeScript/issues/16016#issuecomment-303462193
   */
  init: function(this: any) {
    this.el.addEventListener('loaded', () => {
      console.log('scene loaded');
      const loader = document.getElementById('loader') as HTMLDivElement;
      (loader.parentElement as HTMLDivElement).removeChild(loader);
    })
  }
}
