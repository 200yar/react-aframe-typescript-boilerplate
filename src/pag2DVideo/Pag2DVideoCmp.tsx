//todo: hacer componente con raycaster para que entidad sea clicable (ejemplo: btn play videoplayer)
//todo: usar parte del video pnau going bang
//todo: usar raycaster para interaccion de raton con video player controls

/// <reference path="../index.d.ts"/>
import React from 'react';
import 'aframe-video-controls/dist/aframe-video-controls';
import Loader from "../components/loader/LoaderCmp";
import TopMenu from "../components/topMenu/TopMenuCmp";
import SideMenu from "../components/sideMenu/SideMenuCmp";
import {SIDE_MENU_ITEMS} from "../components/sideMenu/SideMenuItems";
const burgerIcon = require('../components/topMenu/burger-icon.svg');
const playerBtn: any = {width: '50px'};

interface IProps {
  readonly history?: any;
}

export default class Pag2DVideoCmp extends React.PureComponent<IProps> {

  public refs: {loader: Loader, sideMenu: SideMenu};

  public componentDidMount() {
    const assets = document.getElementById('assets') as AFrame.Entity;
    assets && assets.addEventListener('loaded', () => {
      this.refs.loader.hide();
    })
  }

  private onClickBtnPause = () => {
    const video1El = document.querySelector('#video1') as HTMLVideoElement;
    const video2El = document.querySelector('#video2') as HTMLVideoElement;
    video1El.pause();
    video2El.pause();
  };

  private onClickBtnPlay = () => {
    const video1El = document.querySelector('#video1') as HTMLVideoElement;
    const video2El = document.querySelector('#video2') as HTMLVideoElement;
    video1El.play();
    video2El.play();
  };

  private onClickLeftIcon = () => {
    this.refs.sideMenu.show();
  };

  private onClickBurgerIcon = () => {
    this.refs.sideMenu.show();
  };

  public render() {
    return (
      <div>

        <Loader ref="loader">Loader</Loader>

        <SideMenu ref="sideMenu" title="2D/3D Video" items={ SIDE_MENU_ITEMS } itemActive="1" />

        <TopMenu leftIcon={ burgerIcon } onClickMenuBtn={ this.onClickLeftIcon }>
          <a onClick={ this.onClickBtnPlay } className="top-menu-item" style={ playerBtn }>Play</a>
          <a onClick={ this.onClickBtnPause } className="top-menu-item" style={ playerBtn }>Pause</a>
        </TopMenu>

        <a-scene raycaster="far: 100; objects: [src='#video-play-image']; interval: 150" cursor="rayOrigin: mouse">

            <a-assets id="assets">
              <video id="video1"><source type="video/mp4" src="video/echo-hereweare.mp4"/></video>
              <video id="video2"><source type="video/mp4" src="video/canguros.mp4"/></video>
              <img id="video-play-image" src="img/play-video-btn.png"/>
              <img id="video-pause-image" src="img/pause-video-btn.png"/>
            </a-assets>

            <a-camera position="0 0.5 1" reverse-mouse-drag="true"><a-cursor id="cursor" color="yellow"/></a-camera>

            <a-videosphere src="#video2"/>

            <a-entity rotation="0 -45 0">
              <a-video src="#video1" position="0 2.5 -1" scale="2 1 0"/>
              <a-entity video-controls="src:#video1; distance:1; size:2"/>
            </a-entity>

          </a-scene>

      </div>
    )
  }
}
