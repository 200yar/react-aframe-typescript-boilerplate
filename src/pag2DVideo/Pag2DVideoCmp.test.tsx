import 'aframe';
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, ShallowWrapper} from 'enzyme';
import {shallowToJson} from "enzyme-to-json";
import {Pag2DVideoCmp} from "./Pag2DVideoCmp";
Enzyme.configure({ adapter: new Adapter() });

describe('Pag2DVideoCmp Component', () => {

  const component = shallow(<Pag2DVideoCmp>Dummy Content</Pag2DVideoCmp>);

  it("is defined", () => {
    expect(component).toBeDefined();
  });

  it('renders without crashing', () => {
    expect(component).toBeInstanceOf(ShallowWrapper);
  });

  it('matches snapshot', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });

});
