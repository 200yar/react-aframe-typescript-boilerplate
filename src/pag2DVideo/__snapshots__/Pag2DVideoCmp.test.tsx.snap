// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pag2DVideoCmp Component matches snapshot 1`] = `
<div>
  <Loader>
    Loading
  </Loader>
  <SideMenu
    itemActive="1"
    items={
      Array [
        Object {
          "active": false,
          "ico": "start.svg",
          "name": "Home",
          "url": "#/",
        },
        Object {
          "active": false,
          "ico": "video-player.svg",
          "name": "2D/3D Video",
          "url": "#/2dvideo",
        },
        Object {
          "active": false,
          "ico": "360-degrees.svg",
          "name": "360 Video",
          "url": "#/360video",
        },
        Object {
          "active": false,
          "ico": "3d-printer.svg",
          "name": "3D Model Animation",
          "url": "#/3dmodel",
        },
      ]
    }
    title="2D/3D Video"
  />
  <TopMenu
    leftIcon="burger-icon.svg"
    onClickMenuBtn={[Function]}
  >
    <a
      className="top-menu-item"
      onClick={[Function]}
      style={
        Object {
          "width": "50px",
        }
      }
    >
      Play
    </a>
    <a
      className="top-menu-item"
      onClick={[Function]}
      style={
        Object {
          "width": "50px",
        }
      }
    >
      Pause
    </a>
  </TopMenu>
  <a-scene
    cursor="rayOrigin: mouse"
    raycaster="far: 100; objects: [src='#video-play-image']; interval: 150"
  >
    <a-assets
      id="assets"
    >
      <video
        id="video1"
        preload="auto"
      >
        <source
          src="video/echo-hereweare.mp4"
          type="video/mp4"
        />
      </video>
      <video
        id="video2"
        preload="auto"
      >
        <source
          src="video/canguros.mp4"
          type="video/mp4"
        />
      </video>
      <img
        id="video-play-image"
        src="img/play-video-btn.png"
      />
      <img
        id="video-pause-image"
        src="img/pause-video-btn.png"
      />
    </a-assets>
    <a-camera
      position="0 0.5 1"
      reverse-mouse-drag="true"
    >
      <a-cursor
        color="yellow"
        id="cursor"
      />
    </a-camera>
    <a-videosphere
      src="#video2"
    />
    <a-entity
      rotation="0 -45 0"
    >
      <a-video
        position="0 2.5 -1"
        scale="2 1 0"
        src="#video1"
      />
      <a-entity
        video-controls="src:#video1; distance:1; size:2"
      />
    </a-entity>
  </a-scene>
</div>
`;
